### PR TITLE
update sp1 and risc0

### LINKED ISSUE #1205: Operator: Bump SP1 and Risc0 dependencies
We currently use sp1 = v1.0.1 and risc0 = v1.0.1 we should bump these to the latest version, and make the operator work with the old one until the next upgrade

### PR DESCRIPTION
**Pull request by PatStiles**
# Update SP1 and Risc0 verifier versions

closes #1205 

# This PR:
 - Bumps the `risc0` and `sp1` verifier versions of operators. 
 - Provides retro-compatibility for each version. If proof verification for SP1 and Risc0 between the current and old ('_old') verifier versions fails within the operator it will try verifying using the older version of SP1/Risc0 before returning false.
 - Changes the prompt library in the quiz example from `dialoguer` -> `inquire` to accomodate a dependency conflict.

### Verify network bindings works on a local network:
- `make test_risc_zero_go_bindings_macos`
- `make test_sp1_go_bindings_macos`
- run local testnet and confirm you can submit sp1 and risc0 proofs. via
`make batcher_send_risc0_task`
`make batcher_send_sp1_task`
- confirm examples work:
`cd examples/zkquiz && make answer_quix`
`cd examples/validating-public-input && make generate_risc_zero_fibonacci_proof`

### Verify Retro Compatiblility of Operator Verifiers:
To verify operator retro-compatibility we deploy the rest of infrastructure of Aligned and boot the retro-compatible operator. This tests that network retro-compatible operator can verify SP1/Risc0 proofs from both verifier versions.

Note:
Commands suffixed with `_old` indicate the deprecated version of the SP1/Risc0 verifier that is currently running on `testnet`

To test this have two repos of aligned_layer one set to `testnet` and one to `1205-bump-sp1-and-risc0-version`. 

In `testnet` repo:
```
go clean
go mod tidy
make go_deps
make deps
```
- Then you must run anvil, aggregator and batcher:
```
make anvil_start_with_block_time
```
```
make aggregator_start
```
```
make batcher_start_local
```
 
 In `1205-bump-sp1-and-risc0-version` repo:
 ```
make deps
```
- Then you can run operator
```
make operator_register_and_start
```
- Send proof from the older SP1 and Risc0 versions from the `testnet` repo.
```
make batcher_send_burst_groth16
make batcher_send_sp1_burst
make batcher_send_risc0_burst
```
Observe the `Risc0 proof verification failed.` and `SP1 proof verification failed.` logs are emitted and the operator successfully verifying the proofs from the old SP1/Risc0 verifiers. If you send a proof from the latest version of SP1/Risc0 you should notice its verification fails in the batcher.

- Keep the operator running on the `1205-bump-sp1-and-risc0-version` repo.
- In the `testnet` repo, kill the batcher. 
- Navigate to the `1205-bump-sp1-and-risc0-version` and start the batcher within the repo.
- Send proof from both the SP1 and Risc0 versions from the `1205-bump-sp1-and-risc0-version` repo.
```
make batcher_send_sp1_burst
make batcher_send_risc0_burst
```
Observe the operator successfully verifying the proofs from the new SP1/Risc0 verifiers. If you send a proof from the older version of SP1/Risc0 you should notice its verification fails in the batcher.

## Type of change

Please delete options that are not relevant.

- [x] New feature

## Checklist
- [ ] Linked to Github Issue
- [ ] This change depends on code or research by an external entity
  - [ ] Acknowledgements were updated to give credit
- [ ] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] This change is an Optimization
  - [ ] Benchmarks added/run
- [ ] Has a known issue
  - [Link to the open issue addressing it]()

### CODE REVIEW

**Review by MarcosNicolau — APPROVED** [review]
Worked for me!

File: Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ ifeq ($(OS),Darwin)
 endif
 
 ifeq ($(OS),Linux)
-	LD_LIBRARY_PATH += $(CURDIR)/operator/risc_zero/lib
+	export LD_LIBRARY_PATH+=$(CURDIR)/operator/risc_zero_old/lib:$(CURDIR)/operator/risc_zero/lib
```
Comment (entropidelic) [inline Makefile]: this should not have the `export` keyword

File: operator/risc_zero/.gitignore [hunk 2]
```diff
@@ -1,2 +1,2 @@
-lib/librisc_zero_verifier_ffi.a
-lib/librisc_zero_verifier_ffi.dylib
+lib/librisc_zero_verifier_ffi.so
+lib/librisc_zero_verifier.dylib
```
Comment (entropidelic) [inline operator/risc_zero/.gitignore:2]: were these changes necessary?

File: operator/sp1/.gitignore [hunk 3]
```diff
@@ -1,2 +1,2 @@
-lib/libsp1_verifier.a
+lie/libsp1_verifier.so
```
Comment (entropidelic) [inline operator/sp1/.gitignore]: typo

**Comment by Oppen** [comment]
I believe there might be missing steps in the testing procedure:
```
Welcome to the zkQuiz! Answer questions, generate a zkProof, and claim your NFT!
error: the following required arguments were not provided:
  --keystore-path <KEYSTORE_PATH>
  --verifier-contract-address <VERIFIER_CONTRACT_ADDRESS>
Usage: quiz-script --keystore-path <KEYSTORE_PATH> --verifier-contract-address <VERIFIER_CONTRACT_ADDRESS>              
```

File: Makefile (hunk 1 above)
Comment (Oppen) [reply Makefile]: Why not? Isn't the point to have this on the environment for child processes?

Comment (entropidelic) [reply Makefile]: we had a bug related to this previosly which was solved in https://github.com/yetanotherco/aligned_layer/pull/1299. Maybe the problem was another?

File: operator/risc_zero/.gitignore (hunk 2 above)
Comment (PatStiles) [reply operator/risc_zero/.gitignore:2]: I removed the `_ffi` suffix initially as I found it redundant. When compiling on linux the linker was unable to find the shared library for risc0 without having the `_ffi` suffix added which leads to the difference in naming between the `.so` and `.dylib`. I was not sure why we initially were ignoring the `.a` file so I changed to ignoring the `.so`. I don't have a strong opinion on the naming for the macos shared library.

**Comment by PatStiles** [comment]
@Oppen apologies corrected the example commands in the description above.

File: operator/sp1/.gitignore (hunk 3 above)
Comment (PatStiles) [reply operator/sp1/.gitignore]: Fixed!

File: docs/3_guides/6_setup_aligned.md [hunk 4]
```diff
@@ -493,7 +493,7 @@ make install_aligned_compiling
 
 The SP1 and Risc0 proofs need the proof file and the vm program file.
 The current SP1 version used in Aligned is
-`v1.0.1` and the current Risc0 version used in Aligned is v1.0.1.
+`v2.0.0` and the current Risc0 version used in Aligned is `v1.1.2`.
```
Comment (JuArce) [inline docs/3_guides/6_setup_aligned.md]: Isn't it v3.0.0-rc3 instead of v2.0.0

**Review by uri-99 — APPROVED** [review]

**Review by JuArce — CHANGES_REQUESTED** [review]
Close
